src/Sylius/Component/Core/spec/Promotion/Checker/ContainsTaxonRuleCheckerSpec.php

```diff
@@ -0,0 +1,156 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Component\Core\Promotion\Checker;
+
+use PhpSpec\ObjectBehavior;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Sylius\Component\Core\Model\TaxonInterface;
+use Sylius\Component\Core\Promotion\Checker\ContainsTaxonRuleChecker;
+use Sylius\Component\Promotion\Checker\RuleCheckerInterface;
+use Sylius\Component\Promotion\Model\PromotionSubjectInterface;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
+
+/**
+ * @mixin ContainsTaxonRuleChecker
+ *
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class ContainsTaxonRuleCheckerSpec extends ObjectBehavior
+{
+    function let(TaxonRepositoryInterface $taxonRepository)
+    {
+        $this->beConstructedWith($taxonRepository);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Component\Core\Promotion\Checker\ContainsTaxonRuleChecker');
+    }
+
+    function it_implements_rule_checker_interface()
+    {
+        $this->shouldImplement(RuleCheckerInterface::class);
+    }
+
+    function it_returns_true_if_order_has_required_number_of_products_from_configured_taxon(
+        OrderInterface $order,
+        OrderItemInterface $compositeBowItem,
+        OrderItemInterface $longbowItem,
+        ProductInterface $compositeBow,
+        ProductInterface $longbow,
+        TaxonInterface $bows,
+        TaxonRepositoryInterface $taxonRepository
+    ) {
+        $order
+            ->getItems()
+            ->willReturn(new \ArrayIterator([$compositeBowItem->getWrappedObject(), $longbowItem->getWrappedObject()]))
+        ;
+
+        $taxonRepository->findOneBy(['code' => 'bows'])->willReturn($bows);
+
+        $compositeBowItem->getProduct()->willReturn($compositeBow);
+        $compositeBow->hasTaxon($bows)->willReturn(true);
+        $compositeBowItem->getQuantity()->willReturn(4);
+
+        $longbowItem->getProduct()->willReturn($longbow);
+        $longbow->hasTaxon($bows)->willReturn(true);
+        $longbowItem->getQuantity()->willReturn(5);
+
+        $this->isEligible($order, ['taxon' => 'bows', 'count' => 5])->shouldReturn(true);
+    }
+
+    function it_returns_false_if_order_has_not_required_number_of_products_from_configured_taxon(
```

Comment:
`*_the_order_doesnt_have_the_required_*`
or `an_order`, no idea if there was a discussion about treating it as one order, or any given that may come by